### DISCUSSION TITLE
Welcome to vaceline Discussions!

### DISCUSSION THREAD

**Discussion by shqld** [shqld/vaceline#16] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by beatak** [comment]
I have a local branch that helps a case like `return synth(404, "not found");` in pure VCL, which is usually done like `return 404 "not found"` in Fastly-VCL. In order to do this, I'm implementing a new AST node called `ReturnExpressionStatement`. There are still rough edges, so it might take a while.

The other thing I want to do is to improve in `loc` object of each node? I haven't dug in too much, but it seems to me its `.offset` seems short sometimes. For example, a `loc` object of `IfStatement`, if you substring by the `start.offset` and `end.offset`, it almost always seems to miss the closing bracket.

Oh the one last thing 😸 THANK YOU for this project. It helps me a lot!

**Reply by shqld** [reply]
Thanks a lot for joining this discussion @beatak!! Firstly, I'm so sorry for this late reply (I have to check the notification status somehow.)

> I have a local branch that helps a case like return synth(404, "not found"); in pure VCL, which is usually done like return 404 "not found" in Fastly-VCL. In order to do this, I'm implementing a new AST node called ReturnExpressionStatement. There are still rough edges, so it might take a while.

Thank you for improving vaceline. I can do that in place of you or would like to help you, so please tell me how it's going on!

> The other thing I want to do is to improve in loc object of each node? I haven't dug in too much, but it seems to me its .offset seems short sometimes. For example, a loc object of IfStatement, if you substring by the start.offset and end.offset, it almost always seems to miss the closing bracket.

I'm gonna looking into this issue now 👀
